vilebot/src/main/java/com/oldterns/vilebot/handlers/user/News.java

```diff
@@ -0,0 +1,140 @@
+            newsFeedsByCategory.put( "sudbury", new URL( "https://rss.cbc.ca/lineup/canada-sudbury.xml" ) );
+            newsFeedsByCategory.put( "windsor", new URL( "https://rss.cbc.ca/lineup/canada-windsor.xml" ) );
+            newsFeedsByCategory.put( "london", new URL( "https://www.cbc.ca/cmlink/rss-canada-london" ) );
+            newsFeedsByCategory.put( "waterloo", new URL( "https://rss.cbc.ca/lineup/canada-kitchenerwaterloo.xml" ) );
+            newsFeedsByCategory.put( "toronto", new URL( "https://rss.cbc.ca/lineup/canada-toronto.xml" ) );
+            newsFeedsByCategory.put( "hamilton", new URL( "https://rss.cbc.ca/lineup/canada-hamiltonnews.xml" ) );
+            newsFeedsByCategory.put( "montreal", new URL( "http://rss.cbc.ca/lineup/canada-montreal.xml" ) );
+            newsFeedsByCategory.put( "nb", new URL( "https://rss.cbc.ca/lineup/canada-newbrunswick.xml" ) );
+            newsFeedsByCategory.put( "pei", new URL( "https://rss.cbc.ca/lineup/canada-pei.xml" ) );
+            newsFeedsByCategory.put( "ns", new URL( "https://rss.cbc.ca/lineup/canada-novascotia.xml" ) );
+            newsFeedsByCategory.put( "newfoundland", new URL( "https://rss.cbc.ca/lineup/canada-newfoundland.xml" ) );
+            newsFeedsByCategory.put( "north", new URL( "https://rss.cbc.ca/lineup/canada-north.xml" ) );
+        }
+        catch ( MalformedURLException e )
+        {
+            logger.error( "Error loading news URLs" );
+            throw new RuntimeException( e );
+        }
+    }
+
+    private static final Pattern NEWS_PATTERN = Pattern.compile( "^!news(?: ([a-zA-Z]{2,})|)" );
+
+    @Override
+    public void onGenericMessage( final GenericMessageEvent event )
+    {
+        String text = event.getMessage();
+        Matcher matcher = NEWS_PATTERN.matcher( text );
+
+        if ( matcher.matches() )
+        {
+            currentNews( event, matcher );
+        }
+    }
+
+    private void currentNews( GenericMessageEvent event, Matcher matcher )
+    {
+        String category = matcher.group( 1 ); // The news category
+        if ( category == null )
+        {
+            category = defaultCategory;
+        }
+        category = category.toLowerCase();
+
+        if ( newsFeedsByCategory.containsKey( category ) )
+        {
+            printHeadlines( event, category );
+        }
+        else
+        {
+            event.respondWith( "No news feed available for " + category );
+        }
+    }
+
+    private void printHeadlines( GenericMessageEvent event, String category )
+    {
+        SyndFeedInput input = new SyndFeedInput();
+        SyndFeed feed = null;
+        try
+        {
+            feed = input.build( new XmlReader( newsFeedsByCategory.get( category ) ) );
+        }
+        catch ( FeedException | IOException e )
+        {
+            logger.error( e.getMessage() );
+            logger.error( "Error opening RSS feed" );
```

Comment:
consider responding to the event with this error message as well so the user knows why they're not receiving headlines

Reply:
Noted.